### ISSUE TITLE
trouble with https feed

### ISSUE THREAD

**Issue by deep-42-thought** [embyt/balloonrss#1]
Hi,
I have trouble accessing a feed via https:// - using http:// for the same feed works just fine (but I'd prefer to not host the feed via http, because the URI contains credentials to accessing the feed, which I'd like to not make public).
The certificate chain's root is Let's Encrypt. Reading a little in stackoverflow, it seems to point to invalid certificates and -chains, but the site works well in a browser. Is there anything else, that must be done to make a certificate work with .NET?
For reference: The tech-stack of the feed is nginx with a certbot-provided certificate (pretty much standard, if you ask me).

**Comment by romor** [comment]
Not much I can help on this as I do not use SSL feeds and never ran in this issue.

Some directions I can give:

The relevant code is `RetrieveChannel()` in Retriever.cs:119ff: 
https://github.com/embyt/balloonrss/blob/5478ba961e719974b285987a4dbe457f3beb1fb8/src/Retriever.cs#L127-L138

What could also be relevant:
https://stackoverflow.com/questions/41577093/webrequest-error-could-not-create-ssl-tls-secure-channel

Hope this helps.

**Comment by deep-42-thought** [comment]
I tried around a little further, yesterday, and noticed, that the project is still on .NET framework 3.5 (I tried to compile on linux with .NET Core 6). Looking into the stackoverflow post you linked, I get the feeling, that .NET Framework 3.5 has a quite low default cipher (and probably cannot even speak TLS 1.2 or TLS 1.3 - does this sound reasonable?
Would it be possible for you to migrate to a more recent version of .NET? Then I could look into actually modifying the code :)

**Comment by romor** [comment]
Yes, it would make sense to update the code to a current version of .NET.
Not sure, however, when I can spend some time on this. But I will put it on my list...

**Comment by deep-42-thought** [comment]
Thank you for considering this. No hurry, I know, how time consuming these "side projects" (in sum) can be.

**Comment by romor** [comment]
I finally found some time to migrate the project to .NET Core 7.0.
However, I did not migrate the used and deprecated `WebRequest` to `HttpClient` yet. Feel free, if you want to contribute.
